src/popup/components/InputAmount.vue

```diff
@@ -1,82 +1,88 @@
 <template>
   <InputField
-    v-validate="ownValidation ? validation : {
+    v-validate="{
       required: true,
       min_value_exclusive: 0,
-      ...+balance.minus(fee) > 0 ? { max_value: max } : {},
       enough_ae: fee.toString(),
-      ...validation,
+      ...max > 0 ? { max_value: max } : {},
     }"
+    v-bind="$attrs"
     name="amount"
-    :message="$attrs['message'] || errors.first('amount')"
     class="input-amount"
     type="number"
-    v-bind="$attrs"
     placeholder="0.00"
+    :value="value"
+    :message="$attrs['message'] || errors.first('amount')"
     :label="$attrs.label || $t('pages.tipPage.amountLabel')"
     @input="$emit('input', $event)"
   >
     <template #after>
-      <span class="token">AE</span>
+      <span class="token">{{ AETERNITY_SYMBOL }}</span>
       <span
         class="amount"
         data-cy="amount-currency"
       >
-        {{ `(${formatCurrency(currencyAmount)})` }}
+        ({{ currencyAmount }})
       </span>
     </template>
   </InputField>
 </template>
 
-<script>
-import { mapGetters } from 'vuex';
-import { pick } from 'lodash-es';
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  onMounted,
+  ref,
+  watch,
+} from '@vue/composition-api';
 import { SCHEMA } from '@aeternity/aepp-sdk';
 import BigNumber from 'bignumber.js';
-import { calculateFee } from '../utils';
-import { useSdk } from '../../composables';
+import { AETERNITY_SYMBOL, calculateFee, rxJsObservableToVueState } from '../utils';
+import { useGetter, useSdk } from '../../composables';
 import InputField from './InputField.vue';
 
-export default {
+export default defineComponent({
   components: {
     InputField,
   },
   props: {
+    value: { type: [String, Number], default: '' },
     noToken: { type: Boolean },
-    ownValidation: { type: Boolean },
-    validation: { type: Object, default: null },
   },
-  data() {
+  setup(props, { emit, root }) {
+    const { getSdk } = useSdk();
+
+    const fee = ref(new BigNumber(0));
+
+    const balance = rxJsObservableToVueState<BigNumber>(
+      (root.$store.state as any).observables.balance,
+    );
+
+    const formatCurrency = useGetter('formatCurrency');
+    const currentCurrencyRate = useGetter('currentCurrencyRate');
+    const hasError = computed(() => (root as any).$validator.errors.has('amount'));
+    const max = computed(() => balance.value.minus(fee.value).toNumber());
+    const currencyAmount = computed(
+      () => formatCurrency.value((+(props.value || 0) * currentCurrencyRate.value).toFixed(2)),
+    );
```

Comment:
As mentioned before - you could use `convertToCurrencyFormatted`

Reply:
Thanks, makes sense. Changed.